### PR TITLE
chore: removed tsd and type tests

### LINKED ISSUE #922: [chore] Remove `tsd` library and type tests
### Description

Removes the tsd library and all the type tests from the graphql-hooks package. Reason for removing is that hte whole library was re-written to use Typescript and the check-types script will check all the types, so running tsd became obsoloete (see the following https://github.com/nearform/graphql-hooks/pull/799 and https://github.com/nearform/graphql-hooks/pull/799#discussion_r833635348).

Related PR:
https://github.com/nearform/graphql-hooks/pull/919

### PR DESCRIPTION
**Pull request by radomird**
### What does this PR do?

Removes the `tsd` library and all the type tests from the `graphql-hooks` package. Reason for removing is that hte whole library was re-written to use Typescript and the `check-types` script will check all the types, so running `tsd` became obsoloete (see the following [PR](https://github.com/nearform/graphql-hooks/pull/799) and [comment](https://github.com/nearform/graphql-hooks/pull/799#discussion_r833635348)).

### Related issues

https://github.com/nearform/graphql-hooks/pull/919

Closes #922

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,7 +9,6 @@
     "build:packages": "lerna run build --no-private",
     "build:site": "lerna run build --scope=create-react-app",
     "postinstall": "lerna bootstrap --hoist --no-ci && npm run build:packages",
-    "tsd": "lerna run tsd",
```
Comment (simoneb) [inline package.json:12]: this command assumes that there's a tsd script in at least some of the monorepo package.json files. we should remove it from there too I guess

Comment (radomird) [reply package.json:12]: Yes, I've checked all of them and there weren't in any of the packages...

Comment (simoneb) [reply package.json:12]: then this command was already doing nothing :D https://github.com/nearform/graphql-hooks/actions/runs/2950012855/jobs/4714237058#step:8:12

**Comment by radomird** [comment]
The build fails because of this issue: https://github.com/nearform/graphql-hooks/issues/920.
After resolving that we can proceed with merging this PR.

**Review by ilteoood — APPROVED** [review]

**Pull request merged**